src/ethereum/frontier/vm/instructions/bitwise.py

```diff
@@ -0,0 +1,148 @@
+"""
+Ethereum Virtual Machine (EVM) Bitwise Instructions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. contents:: Table of Contents
+    :backlinks: none
+    :local:
+
+Introduction
+------------
+
+Implementations of the EVM bitwise instructions.
+"""
+
+from ethereum.base_types import U256
+
+from .. import Evm
+from ..gas import GAS_VERY_LOW, subtract_gas
+from ..stack import pop, push
+
+
+def bitwise_and(evm: Evm) -> None:
+    """
+    Bitwise AND operation of the top 2 elements of the stack. Pushes the
+    result back on the stack.
+
+    Parameters
+    ----------
+    evm :
+        The current EVM frame.
+
+    Raises
+    ------
+    StackUnderflowError
+        If `len(stack)` is less than `2`.
+    OutOfGasError
+        If `evm.gas_left` is less than `GAS_VERY_LOW`.
+    """
+    evm.gas_left = subtract_gas(evm.gas_left, GAS_VERY_LOW)
+    x = pop(evm.stack)
+    y = pop(evm.stack)
+    push(evm.stack, x & y)
+
+
+def bitwise_or(evm: Evm) -> None:
+    """
+    Bitwise OR operation of the top 2 elements of the stack. Pushes the
+    result back on the stack.
+
+    Parameters
+    ----------
+    evm :
+        The current EVM frame.
+
+    Raises
+    ------
+    StackUnderflowError
+        If `len(stack)` is less than `2`.
+    OutOfGasError
+        If `evm.gas_left` is less than `GAS_VERY_LOW`.
+    """
+    evm.gas_left = subtract_gas(evm.gas_left, GAS_VERY_LOW)
+    x = pop(evm.stack)
+    y = pop(evm.stack)
+    push(evm.stack, x | y)
+
+
+def bitwise_xor(evm: Evm) -> None:
+    """
+    Bitwise XOR operation of the top 2 elements of the stack. Pushes the
+    result back on the stack.
+
+    Parameters
+    ----------
+    evm :
+        The current EVM frame.
+
+    Raises
+    ------
+    StackUnderflowError
+        If `len(stack)` is less than `2`.
+    OutOfGasError
+        If `evm.gas_left` is less than `GAS_VERY_LOW`.
+    """
+    evm.gas_left = subtract_gas(evm.gas_left, GAS_VERY_LOW)
+    x = pop(evm.stack)
+    y = pop(evm.stack)
+    push(evm.stack, x ^ y)
+
+
+def bitwise_not(evm: Evm) -> None:
+    """
+    Bitwise NOT operation of the top element of the stack. Pushes the
+    result back on the stack.
+
+    Parameters
+    ----------
+    evm :
+        The current EVM frame.
+
+    Raises
+    ------
+    StackUnderflowError
+        If `len(stack)` is less than `1`.
+    OutOfGasError
+        If `evm.gas_left` is less than `GAS_VERY_LOW`.
+    """
+    evm.gas_left = subtract_gas(evm.gas_left, GAS_VERY_LOW)
+    x = pop(evm.stack)
+    push(evm.stack, ~x)
+
+
+def get_byte(evm: Evm) -> None:
+    """
+    For a word (defined by next top element of the stack), retrieve the
+    Nth byte (0-indexed and defined by top element of stack) from the left
+    to right (MSB to LSB).
```

Comment:
```suggestion
    For a word (defined by next top element of the stack), retrieve the
    Nth byte (0-indexed and defined by top element of stack) from the left (most significant)
    to right (least significant).
```

Reply:
Makes sense. These changes would be reflecting in #248